platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -14,104 +14,286 @@
 package org.codice.ddf.platform.filter.delegate;
 
 import java.io.IOException;
-import java.util.Collection;
 import java.util.Collections;
-import java.util.Iterator;
-import java.util.LinkedList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.concurrent.CopyOnWriteArraySet;
 
+import javax.annotation.Nonnull;
 import javax.servlet.Filter;
 import javax.servlet.FilterChain;
 import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.filter.SecurityFilter;
 import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
-import org.osgi.framework.Constants;
 import org.osgi.framework.FrameworkUtil;
 import org.osgi.framework.InvalidSyntaxException;
 import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * {@link DelegateServletFilter} is meant to detect any ServletFilters
- * and call their {@link FilterChain}. If none are found it becomes a pass
- * through filter.
- *
+ * {@link DelegateServletFilter} is meant to detect any {@link SecurityFilter}s and doFilter through
+ * those before continuing on its {@link FilterChain}. If no {@link SecurityFilter}s are found, it
+ * becomes a pass-through filter.
  */
 public class DelegateServletFilter implements Filter {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DelegateServletFilter.class);
 
+    private CopyOnWriteArraySet<String> keysOfInitializedSecurityFilters;
```

Comment:
Yes, `SecurityFilter`s are only added or removed when specific bundles are redeployed, which is not common. There are only about 5 `SecurityFIlter`s that are there by default.